src/ol/PluggableMap.js

```diff
@@ -1016,15 +1016,21 @@ class PluggableMap extends BaseObject {
       // coordinates so interactions cannot be used.
       return;
     }
-    const target = /** @type {Node} */ (mapBrowserEvent.originalEvent.target);
     if (!mapBrowserEvent.dragging) {
+      const rootNode = this.viewport_.getRootNode
+        ? this.viewport_.getRootNode()
+        : document;
+      const originalEvent = /** @type {PointerEvent} */ (mapBrowserEvent.originalEvent);
+      const target =
+        rootNode === document
+          ? /** @type {Node} */ (originalEvent.target)
+          : /** @type {ShadowRoot} */ (rootNode).elementFromPoint(
+              originalEvent.clientX,
```

Comment:
This is subtle to have these two conditionals (here and line 1022).  Are there cases where `getRootNode` returns `document` and we don’t want to call `elementFromPoint`?

If not, it seems like the code would be clearer with just one `if` to handle the case where `getRootNode` exists.

Reply:
`getRootNode()` returns `document`, except when in a web component, where it returns the shadow root. The check for `getRootNode` is only needed because Internet Explorer does not have a `getRootNode` function on a `HTMLElement`.

Reply:
Thanks for the explanation.